File: server/models/Tournament.js
const mongoose = require("mongoose");

const tournamentSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      unique: true,
    },

    game: {
      type: String,
      required: true,
      enum: ["BGMI", "PUBG", "FREE FIRE"],
    },

    mode: {
      type: String,
      required: true,
      enum: ["Solo", "Duo", "Squad"],
    },

    entryFee: {
      type: Number,
      required: true,
      default: 0,
    },

    prizePool: {
      type: Number,
      required: true,
      default: 0,
    },

    maxTeams: {
      type: Number,
      default: 25,
    },

    registeredTeams: [
      {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Team",
      },
    ],

    startDate: {
      type: Date,
      required: true,
    },

    status: {
      type: String,
      enum: ["upcoming", "live", "completed"],
      default: "upcoming",
    },

    createdBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
    },
  },
  {
    timestamps: true,
  },
);

module.exports = mongoose.model("Tournament", tournamentSchema);

File: backend/app/schemas/tournament.py
from pydantic import BaseModel, Field
from datetime import datetime
from decimal import Decimal
class TournamentCreate(BaseModel):
    title:str=Field(min_length=1,max_length=150); game:str; mode:str; entryFee:Decimal=Field(default=0,ge=0); prizePool:Decimal=Field(default=0,ge=0); maxTeams:int=Field(default=25,gt=0); startDate:datetime
class TournamentUpdate(BaseModel):
    title:str|None=None; game:str|None=None; mode:str|None=None; entryFee:Decimal|None=Field(default=None,ge=0); prizePool:Decimal|None=Field(default=None,ge=0); maxTeams:int|None=Field(default=None,gt=0); startDate:datetime|None=None
